akka-stream/src/main/scala/akka/stream/javadsl/Graph.scala

```diff
@@ -278,10 +277,10 @@ object FlowGraph {
    * The [[Builder]] is mutable and not thread-safe,
    * thus you should construct your Graph and then share the constructed immutable [[FlowGraph]].
    */
-  def builder(): Builder = new Builder()(new scaladsl.FlowGraph.Builder)
+  def builder[Mat](): Builder[Mat] = new Builder()(new scaladsl.FlowGraph.Builder[Mat])
```

Comment:
MaterializedType would be superfluous, since it is a type, so `M` would be perfectly fine, and then add ScalaDoc explaining that M is the Materialized type?